packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -4242,6 +4244,15 @@ class InputDecorationTheme with Diagnosticable {
   /// The duration of the [InputDecoration.hintText] fade in and fade out animations.
   final Duration? hintFadeDuration;
 
+  /// The maximum number of lines the [InputDecoration.hintText] can occupy.
+  ///
+  /// Defaults to null, which means that the [InputDecoration.hintText] will
+  /// be limited to a single line with [TextOverflow.ellipsis].
```

Comment:
How is that different that `1`? If you include a linebreak in the hint?

Reply:
Yaa, its same behaviour like `1` but just the value is `null`. If we add link break then it will just show first line.

Reply:
I think it should have a default value of 1 actually, right? It looks like InputDecoration.hintMaxLines defaults to TextField.maxLines:

https://github.com/flutter/flutter/blob/31dfefc79e9bc5dd3c0667a0db1ae57ca329a119/packages/flutter/lib/src/material/input_decorator.dart#L3095

Which defaults to 1:

https://github.com/flutter/flutter/blob/31dfefc79e9bc5dd3c0667a0db1ae57ca329a119/packages/flutter/lib/src/material/text_field.dart#L271

So that's why a `null` here ends up giving the same behavior as 1.

Then, can you write two more tests? One that tests the default behavior and one that tests explicitly passing `null`.

Reply:
Yes @justinmc, after this conversation I was looking into TextField and there change is required to use `InputDecorationTheme.hintMaxLines`.